### ISSUE TITLE
structcheck returns false positives for embedded structs

### ISSUE THREAD

**Issue by valyala** [golangci/golangci-lint#826]
Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```bash
$ golangci-lint --version
golangci-lint has version v1.21.0 built from (unknown, mod sum: "h1:HxAxpR8Z0M8omihvQdsD3PF0qPjlqYqp2vMJzstoKeI=") on (unknown)
```

</details>

<details><summary>Config file</summary>

```bash
$ cat .golangci.yml
cat: .golangci.yml: No such file or directory
```

</details>

<details><summary>Go environment</summary>

```bash
$ go version && go env
go version go1.13.1 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/valyala/.cache/go-build"
GOENV="/home/valyala/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/valyala/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/valyala/work/go-src"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/valyala/work/go-src/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/valyala/work/VictoriaMetrics/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build974883207=/tmp/go-build -gno-record-gcc-switches"
```

</details>

See false positives here: https://github.com/VictoriaMetrics/VictoriaMetrics/commit/97ce4e03a57825a783a28e25fc88b9a9e9e09c3c/checks?check_suite_id=269491276

**Comment by kortschak** [comment]
We see the same thing in [gonum/gonum](https://golangci.com/r/github.com/gonum/gonum). Locally, this looks like this:
```
~/src/gonum.org/v1/gonum/graph/encoding/dot [master*]$ golangci-lint run
decode.go:152:2: `graphAttr` is unused (structcheck)
	graphAttr, nodeAttr, edgeAttr encoding.AttributeSetter
	^
```

This field is used embedded.

**Comment by rillig** [comment]
Self-contained example, extracted from https://github.com/rillig/pkglint:

~~~go
package main

import "fmt"

type Autofix struct {
	autofixShortTerm // embedded with implicit name
}

type autofixShortTerm struct {
	actions []string
}

func (a *Autofix) PrintActions() {
	for _, action := range a.actions { // obviously used
		fmt.Println(action)
	}
}

func main() {
	fix := Autofix{}
	fix.actions = append(fix.actions, "action") // obviously used as well
	fix.PrintActions()
}
~~~

Output:
~~~text
$ golangci-lint.exe run
demo.go:10:2: `actions` is unused (structcheck)
        actions []string
~~~

**Comment by bnsblue** [comment]
Encountered this same bug +1

**Comment by jszroberto** [comment]
+1

**Comment by rillig** [comment]
[Still reproducible](https://github.com/golangci/golangci-lint/issues/826#issuecomment-564055536):
~~~text
$ golangci-lint.exe --version
golangci-lint has version v1.45.2 built from (unknown, mod sum: "h1:9I3PzkvscJkFAQpTQi5Ga0V4qWdJERajX1UZ7QqkW+I=") on (unknown)

$ golangci-lint.exe run ./main.go 
main.go:10:2: `actions` is unused (structcheck)
        actions []string
        ^
~~~

**Comment by cliff-comvex** [comment]
Encountered the same issue for embedded structs that says `(... no field or method ..) (typecheck)` For typecheck linter.

**Comment by Antonboom** [comment]
+1

```go
type Event interface {
	eventMarker()
}

type event struct{}         //
func (*event) eventMarker() {}

type SomeEvent struct {
	event
	// ...
}
```

```
field `event` is unused (unused)
```

**Comment by ldez** [comment]
`structcheck` != `unused`

unused is a part of staticcheck https://github.com/dominikh/go-tools/tree/master/unused

**Comment by Antonboom** [comment]
@ldez 

Thanks!
https://github.com/dominikh/go-tools/issues/1312

**Comment by ldez** [comment]
structcheck is deprecated, please stop using it.
